django/contrib/gis/db/backends/base/operations.py

```diff
@@ -104,7 +104,12 @@ def get_geom_placeholder(self, f, value, compiler):
 
     def check_aggregate_support(self, aggregate):
         if aggregate.contains_aggregate == 'gis':
```

Comment:
I think this problem can be solved in a general sense by giving the backends a chance to recursively inspect all expressions. Ansii and I just discussed this on IRC and came up with: https://code.djangoproject.com/ticket/24154

Can you have a look and see if the idea in the ticket will solve the geo-aggregate case? If so, then I'll work on implementing that probably tonight.